### PR TITLE
fix: check for old alpha `.snyk` file format and throw

### PR DESCRIPTION
**Pull request by gclapperton**
#### What does this PR do?

Re-enables throwing of errors when an early alpha format of the `.snyk` file is found. This was temporarily commented out during the switch to Typescript, but was not re-enabled.

### CODE REVIEW

File: test/unit/old-format.test.ts [hunk 1]
```diff
@@ -1,12 +1,9 @@
 import { expect, test } from 'vitest';
 import * as policy from '../../lib';
 
-test('test sensibly bails if gets an old .snyk format', () => {
-  expect(() =>
-    policy.load(__dirname + '/../fixtures/old-snyk-config/').catch((e) => {
-      expect(e.message).toBe('old, unsupported .snyk format detected');
```
Comment (gclapperton) [inline test/unit/old-format.test.ts:7]: An error was still being thrown after the commented out check, but this check did not catch it. Or rather, it did, but then re-threw on the check above (as the errors did not match) and the outer `toThrow` just checked that an error was thrown and not what the error was.

TLDR the test that should have caught the issue was also not working as expected

File: lib/parser/v1.ts [hunk 2]
```diff
@@ -67,11 +67,11 @@ function checkForOldFormat(ruleSet: unknown) {
   // alpha format, and we'll throw
   if (isObject(ruleSet)) {
     for (const id in ruleSet) {
-      if (!Array.isArray(ruleSet[id])) {
+      if (ruleSet[id] && !Array.isArray(ruleSet[id])) {
```
Comment (gclapperton) [inline lib/parser/v1.ts:70]: In #121 we added graceful handling of empty rules, so we must now check that `ruleSet[id]` is not undefined/null

**Review by jcsackett — APPROVED** [review]
Thanks!
